commerce-service/src/main/java/com/liferay/commerce/internal/upgrade/v4_4_0/CommerceOrderAddressUpgradeProcess.java

```diff
@@ -0,0 +1,213 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.internal.upgrade.v4_4_0;
+
+import com.liferay.commerce.model.CommerceOrder;
+import com.liferay.commerce.model.impl.CommerceOrderModelImpl;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.service.ClassNameLocalService;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+/**
+ * @author Alec Sloan
+ */
+public class CommerceOrderAddressUpgradeProcess extends UpgradeProcess {
+
+	public CommerceOrderAddressUpgradeProcess(
+		ClassNameLocalService classNameLocalService) {
+
+		_classNameLocalService = classNameLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		PreparedStatement ps = connection.prepareStatement(
+			"select * from CommerceAddress where classNameId = ?");
+
+		ps.setLong(
+			1, _classNameLocalService.getClassNameId(CommerceOrder.class));
+
+		ResultSet rs = ps.executeQuery();
+
+		// Create columns on CommerceOrder to hold billing and shipping address info for placed orders
+
+		_addOrderAddressColumns();
+
+		// Add to new billing address columns for placed orders
+
+		String updateOrderBillingAddressSQL = StringBundler.concat(
+			"update CommerceOrder set billingName = ?, billingDescription = ?,",
+			"billingStreet1 = ?, billingStreet2 = ?, billingStreet3 = ?,",
+			"billingCity = ?, billingZip = ?, billingRegionId = ?,",
+			"billingCountryId = ?, billingPhoneNumber = ?, billingAddressId = ",
+			"0 where billingAddressId = ?");
```

Comment:
is this going to cause problems -- setting `billingAddressId` and `shippingAddressId` to `0`? Do we assume these values aren't 0 when we use them in the local service or anything?

Reply:
The reason I'm setting those back to 0 is because these addresses belong to the Order model which we're completely removing here so all of those past addressId's won't exist anymore. Although (If I recall correctly) after an order is placed these addressIds will remain the same that they previously were so that we have a history of what model that was used along with the address info. This means that the two can be modified independent of each other.